### ISSUE TITLE
MDBOOK005 orphaned files detection extends beyond book src directory

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#142]
### Description
The MDBOOK005 rule (orphaned files detection) is checking for files outside the mdBook's `src` directory when running the linter. This causes false positives for legitimate documentation files that exist alongside but not within the book.

### Steps to Reproduce

1. Have a project structure like:
```
project/
├── book.toml
├── src/
│   ├── SUMMARY.md
│   └── chapter1.md
└── docs/
    ├── CONFIGURATION_AUDIT.md
    └── src/
        └── chapter2.md
```

2. Run `mdbook-lint lint docs/src`

### Expected Behavior
MDBOOK005 should only check for orphaned files within the mdBook's configured `src` directory (or the directory being linted). Files outside this scope should not be considered.

### Actual Behavior
MDBOOK005 reports warnings for files like `docs/CONFIGURATION_AUDIT.md` that are outside the book's src directory and not intended to be part of the mdBook.

### Impact
Creates noise in the output with false warnings about files that are legitimately outside the book's scope, making it harder to identify actual orphaned files.

### Suggested Fix
1. When running as mdBook preprocessor, only check files within the book's `src` directory
2. When running as CLI, respect the directory boundaries of the paths provided
3. Consider adding a configuration option to control the scope of orphaned file detection

### Environment
- mdbook-lint version: 0.11.1
- Operating System: macOS

### Additional Context
This issue causes confusion when mdbook-lint is used in projects that have multiple documentation locations or when the mdBook is part of a larger documentation structure.

### Proposed Solution Options
1. Add a `--book-root` flag to explicitly set the scope
2. Auto-detect book.toml and use its `[book]` section to determine src directory
3. Add configuration option: `orphaned-files-scope = "book-only"` vs `"recursive"`
